pkg/controller/migmigration/stage.go

```diff
@@ -480,38 +481,126 @@ func (t *Task) stagePodReport(client k8sclient.Client) (report PodStartReport, e
+	report.started = false
 	for _, pod := range podList.Items {
-		ready := false
-		for _, c := range pod.Status.ContainerStatuses {
-			if c.Ready {
-				ready = true
-				break
+		initReady := true
+		for _, c := range pod.Status.InitContainerStatuses {
+			// If the init contianer is waiting, then nothing can happen.
+			if c.State.Waiting != nil {
+				initReady = false
+				report.progress = append(
+				            report.progress,
+				            fmt.Sprintf(
+				                         "Pod %s/%s: Container %s %s",
+				                         pod.Namespace,
+				                         pod.Name,
+				                         c.Name,
+				                         c.State.Waiting.Message))
 			}
-		}
-		if !ready {
-			progress = append(
-				progress,
-				fmt.Sprintf(
-					"Pod %s/%s: Not running. Phase %s",
-					pod.Namespace,
-					pod.Name,
-					pod.Status.Phase))
-			report.started = false
-			if !hasHealthyClaims(&pod) {
-				report.failed = true
+			if c.State.Terminated != nil && c.State.Terminated.ExitCode != 0 {
+				initReady = false
+				report.progress = append(
+					report.progress,
+					fmt.Sprintf(
+						"Pod %s/%s: init container failed to finish",
+						pod.Namespace,
+						pod.Name))
 			}
-		} else {
-			progress = append(
-				progress,
+		}
+		if !initReady {
+			// If init container is not finished or started running warn the user.
+			return
+		}
+
+		// if Pod is running, then move on
+		// pod succeeded phase should never occur for a stage pod
+		if pod.Status.Phase == corev1.PodRunning {
+			report.progress = append(
+				report.progress,
 				fmt.Sprintf(
 					"Pod %s/%s: Running",
 					pod.Namespace,
 					pod.Name))
+			report.started = true
+
+			return
+		}
+
+		// handle pod pending status
+		if pod.Status.Phase == corev1.PodPending {
+			for _, c := range pod.Status.ContainerStatuses {
+				if c.State.Waiting != nil {
+				        report.progress = append(
+				                    report.progress,
+				                    fmt.Sprintf(
+				                         "Pod %s/%s: Container %s %s",
+				                         pod.Namespace,
+				                         pod.Name,
+				                         c.Name,
+				                         c.State.Waiting.Message))
+				}
+			}
+		}
+
+		//TODO: [shurley] eventually, this should cause us to backoff and re-try to the create the pod.
+		if pod.Status.Phase == corev1.PodFailed || pod.Status.Phase == corev1.PodUnknown {
+			report.failed = true
+			report.progress = append(report.progress, report.reasons...)
+			hasHealthyClaims(&pod)
+		}
+	}
+	return
+}
+
+// Match number of stage pods in source and destination cluster
+func (t *Task) allStagePodsMatch() (report []string, err error) {
+	dClient, err := t.getDestinationClient()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	podDList := corev1.PodList{}
 
+	options := k8sclient.MatchingLabels(t.stagePodLabels())
+	err = dClient.List(context.TODO(), options, &podDList)
```

Comment:
question: we seem to be listing stage pods across all namespaces, in order to reduce scope, should we just list in namespaces where the plan in running?